src/Uart/Uart.c

```diff
@@ -420,28 +405,35 @@ Uart_registerErrorHandler(Uart* const uart, const Uart_ErrorHandler handler)
 inline bool
 Uart_getLinkErrors(uint32_t statusRegister, Uart_ErrorFlags* errFlags)
 {
-    if((statusRegister & UART_STATUS_OV) != 0) {
-        errFlags->hasOverrunOccurred = true;
-    }
-    if((statusRegister & UART_STATUS_PE) != 0) {
-        errFlags->hasParityErrorOccurred = true;
-    }
-    if((statusRegister & UART_STATUS_FE) != 0) {
-        errFlags->hasFramingErrorOccurred = true;
-    }
-    if((statusRegister & UART_STATUS_RF) != 0) {
-        errFlags->hasRxFifoFullErrorOccurred = true;
-    }
-    if(errFlags->hasOverrunOccurred || errFlags->hasParityErrorOccurred
-       || errFlags->hasFramingErrorOccurred
-       || errFlags->hasRxFifoFullErrorOccurred) {
-        return true;
+    bool result = false;
+
+    errFlags->hasOverrunOccurred = Uart_getFlag(statusRegister, UART_OV);
+    errFlags->hasParityErrorOccurred = Uart_getFlag(statusRegister, UART_PE);
+    errFlags->hasFramingErrorOccurred = Uart_getFlag(statusRegister, UART_FE);
+    errFlags->hasRxFifoFullErrorOccurred = Uart_getFlag(statusRegister, UART_RF);
+
+    if (errFlags->hasOverrunOccurred ||
+        errFlags->hasParityErrorOccurred ||
+        errFlags->hasFramingErrorOccurred ||
+        errFlags->hasRxFifoFullErrorOccurred) {
+        result = true;
     }
-    return false;
+    
+    return result;
 }
 
-inline uint32_t
-Uart_getStatusRegister(const Uart* const uart)
+bool
+Uart_getFlag(const uint32_t uartRegister, const uint32_t flag)
 {
-    return uart->reg->status;
+    return (bool) ((uartRegister >> flag) & FLAG_MASK);
+}
+
+void
+Uart_setFlag(volatile uint32_t *const uartRegister, const bool set, const uint32_t flag)
```

Comment:
isSet would be better.

Reply:
const bool isSet -> indicating a parameter stating whether the flag is(to be)set. There is less of a problem with enabled/disabled, but set is -> to set, is set, and a set of something.